mkosi/__init__.py

```diff
@@ -7638,6 +7633,11 @@ def run_shell_cmdline(args: MkosiArgs, pipe: bool = False, commands: Optional[Se
     else:
         cmdline += nspawn_rlimit_params()
 
+        # Redirecting output correctly when not running directly from the terminal.
+        console_arg = f"--console={'interactive' if not pipe else 'pipe'}"
```

Comment:
Doesn't this make us fall back to `--console=read-only` in the `boot` case when run in CI?